Tests/SwiftParserTest/translated/OperatorDeclTests.swift

```diff
@@ -302,51 +300,91 @@ final class OperatorDeclTests: XCTestCase {
       infix operator --- : 1️⃣;
       """,
       diagnostics: [
-        // TODO: Old parser expected error on line 3: expected precedence group name after ':' in operator declaration
-        DiagnosticSpec(message: "expected identifier in operator"),
+        DiagnosticSpec(message: "expected precedence group in operator declaration"),
       ]
     )
   }
 
-  func testOperatorDecl15() {
+  func testOperatorDecl15a() {
     AssertParse(
       """
       precedencegroup 1️⃣{ 
         associativity: right
       }
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "expected name in precedencegroup"),
+      ]
+    )
+  }
+
+  func testOperatorDecl15b() {
+    AssertParse(
+      """
       precedencegroup A {
-        associativity 2️⃣right 
+        associativity 1️⃣right
       }
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "expected ':' in 'associativity' property of precedencegroup"),
+      ]
+    )
+  }
+
+
+  func testOperatorDecl15c() {
+    AssertParse(
+      """
       precedencegroup B {
-        3️⃣precedence 123 
+        1️⃣precedence 123
       }
+      """,
+      diagnostics: [
+        DiagnosticSpec(message: "unexpected code 'precedence 123' in precedencegroup"),
+      ]
+    )
+  }
+
+
+  func testOperatorDecl15d() {
+    AssertParse(
+      """
       precedencegroup C {
-        associativity: sinister 
+        associativity: sinister
       }
+      """
+    )
+  }
+
+
+  func testOperatorDecl15e() {
+    AssertParse(
+      """
       precedencegroup D {
-        assignment: 4️⃣no 
```

Comment:
Worth having a fix-it here to cover common cases? no/NO/FALSE/0/etc

Reply:
Honestly, I don’t think so. If you are declaring precedence groups but haven’t understood how Swift specifies booleans, I don’t care about you, I think.